### ISSUE TITLE
On Github Action the code-style check passes, but I get many (correct) errors

### ISSUE THREAD

**Issue by v1kko** [GEMDAT-repos/GEMDAT#132]
We should both fix these errors, and figure out why the Github Action does not catch them

```bash
src/gemdat/transitions.py:137: error: Incompatible return value type (got "dict[int, int]", expected "ndarray[Any, Any]")  [return-value]
src/gemdat/transitions.py:164: error: Argument 1 to "Transitions" has incompatible type "**dict[str, object]"; expected "ndarray[Any, Any]"  [arg-type]
src/gemdat/transitions.py:164: error: Argument 1 to "Transitions" has incompatible type "**dict[str, object]"; expected "int"  [arg-type]
src/gemdat/sites.py:132: error: Function "numpy.core.multiarray.array" is not valid as a type  [valid-type]
src/gemdat/sites.py:132: note: Perhaps you need "Callable[...]" or a callback protocol?
src/gemdat/sites.py:138: error: Function "numpy.core.multiarray.array" is not valid as a type  [valid-type]
src/gemdat/sites.py:138: note: Perhaps you need "Callable[...]" or a callback protocol?
src/gemdat/sites.py:153: error: Argument "occupancy" to "_calculate_site_occupancy" has incompatible type "ndarray[Any, Any]"; expected "dict[int, int]"  [arg-type]
tests/integration/sites_test.py:15: error: "object" not callable  [operator]
tests/integration/rdf_test.py:4: error: "object" not callable  [operator]
src/gemdat/volume.py:350: error: No overload variant of "ravel_multi_index" matches argument types "ndarray[Any, Any]", "tuple[int, ...]"  [call-overload]
src/gemdat/volume.py:350: note: Possible overload variants:
src/gemdat/volume.py:350: note:     def ravel_multi_index(multi_index: Sequence[bool | bool_ | int | integer[Any]], dims: Sequence[SupportsIndex], mode: Literal['raise', 'wrap', 'clip'] | tuple[Literal['raise', 'wrap', 'clip'], ...] = ..., order: Literal['C', 'F'] | None = ...) -> signedinteger[Any]
src/gemdat/volume.py:350: note:     def ravel_multi_index(multi_index: Sequence[_SupportsArray[dtype[bool_ | integer[Any]]] | _NestedSequence[_SupportsArray[dtype[bool_ | integer[Any]]]] | bool | int | _NestedSequence[bool | int]], dims: Sequence[SupportsIndex], mode: Literal['raise', 'wrap', 'clip'] | tuple[Literal['raise', 'wrap', 'clip'], ...] = ..., order: Literal['C', 'F'] | None = ...) -> ndarray[Any, dtype[signedinteger[Any]]]
src/gemdat/volume.py:354: error: Incompatible return value type (got "Volume", expected "ndarray[Any, Any]")  [return-value]
tests/integration/trajectory_test.py:17: error: "object" not callable  [operator]
tests/integration/trajectory_test.py:56: error: "object" not callable  [operator]
tests/integration/trajectory_test.py:75: error: "object" not callable  [operator]
src/gemdat/legacy.py:125: error: "ndarray[Any, Any]" has no attribute "to_vasp_volume"  [attr-defined]
src/gemdat/dashboard/run.py:96: error: Slice index must be an integer, SupportsIndex or None  [misc]
src/gemdat/dashboard/run.py:122: error: Argument 1 to "load_known_material" has incompatible type "str | None"; expected "str"  [arg-type]
src/gemdat/dashboard/run.py:122: error: Argument "supercell" to "load_known_material" has incompatible type "tuple[int | float, int | float, int | float]"; expected "tuple[int, int, int] | None"  [arg-type]
Found 17 errors in 8 files (checked 36 source files)
```

**Comment by stefsmeets** [comment]
I get the same locally. If I run mypy 1.3.0 directly I get 18 errors on main, but through pre-commit it finds 0 problems.

**Comment by stefsmeets** [comment]
pre-commit runs `mypy` in isolation, i.e. on just the source code. If I run it locally, it also follows the imports. I can reproduce the problem like this:

```bash
python -m virtualenv .venv-mypy
pip install mypy==1.3.0
mypy src
# Success: no issues found in 22 source files
```

**Comment by v1kko** [comment]
That makes a lot of sense, lets see if there is a solution for this

**Comment by v1kko** [comment]
fixed in PR #131
